### DISCUSSION TITLE
Livewire seems not to bind data for some unclear reasons.

### DISCUSSION THREAD

**Discussion by wordxpression** [livewire/livewire#7323] [1. Help] [answered]
I tried to use databinding on forms in JetStream, but for some (to me) mysterious reasons it didn't work. So I went back to a complete clean installation of Laravel, added Jetstream to it to be sure there were no JS conflicts and nothing else.

Next I created one component. 'HelloComponent'.

Componentcode in the file 'HelloComponent.php' :
```
<?php

namespace App\Livewire;

use Livewire\Component;

class HelloComponent extends Component
{
    public $hello = "Hello";
    public function render()
    {
        return view('livewire.hello-component');
    }
}
```

Code in the hello.component.blade.php file : 
```
<div>
    <form>
       <input wire:model='hello'/>
       {{ $hello }}
   </form>
</div>
```

I included the component with 
@livewire('hello-component') in the existing 'dashboard' view of JetStream. 

When I change the text in the input, the text in the $hello variable does not change. 

When I change @livewire('hello-component')  into <x-livewire:hello-component/> I get an error message "Unable to locate a class or view for component [livewire:hello-component]."

So to be sure it was nothing in JetStream, I went back to the basics. I built two 'beginners examples', both I made some time ago too when I was still using Livewire 2. Back then they worked. First the counter example from the tutorial. Worked like a charm. Then the second one, Again something like the Hello Component, directly taken from the video tutorial. 

```
<?php
//HelloWorld.php
namespace App\Livewire;

use Livewire\Component;

class HelloWorld extends Component
{
    public $name = 'You';
    public function render()
    {
        return view('livewire.hello-world');
    }
}


```

And hello-world.php
```
<div>
    <input wire:model="name" type="text"/>
    {{ $name }}
</div>
```

And the web.php route file
```
<?php

use Illuminate\Support\Facades\Route;
use App\Livewire\Counter;
use App\Livewire\HelloWorld;

/*
|--------------------------------------------------------------------------
| Web Routes
|--------------------------------------------------------------------------
|
| Here is where you can register web routes for your application. These
| routes are loaded by the RouteServiceProvider and all of them will
| be assigned to the "web" middleware group. Make something great!
|
*/

Route::get('/', function () {
    return view('welcome');
});


 
Route::get('/counter', Counter::class);
Route::get('/hello', HelloWorld::class);
```

That one again did not work. 

The scripts and stylesheet are loaded (otherwise the counter would not have worked either)

What am I doing wrong here?

**Comment by joshhanley** [comment]
@wordxpression All looks ok to me. Do you have a demo repo with your hello world example in it that you can share?

**Reply by wordxpression** [reply]
Not yet, I'll take care of it tomorrow... today not having the opportunity...

**Comment by WilliamPinto-Olmos** [comment, marked as answer]
did you tried adding `.live` to the input binding? Like `wire:model.live="hello"`

Assuming you are in V3, now all input bindings are `defer` by default, so it wont update the variable until you make a proper request to the server.

**Reply by HmHasan22** [reply]
It's working! Thank You

**Comment by pronode** [comment]
Shouldn't that be `hello-world.blade.php`, not `hello-world.php`?

**Reply by whousehomestay** [reply]
And hello.component.blade.php must be hello-component.blade.php right?

**Comment by wordxpression** [comment]
@whousehomestay and @pronode you both were right, I entered the wrong filenames, the original filesnames were as they were supposed to be, excuses for the confusion.

@WilliamPinto-Olmos It was indeed the .live addition which caused the problem. Thank for pointing out. Now it works like a charm again.

Reminds me next time with a major update, I need to read the update notes more carefully. Thanks for the help.

**Reply by pronode** [reply]
Good you got it working!
